### PR TITLE
Mukil/supercluster

### PR DESCRIPTION
**Pull request by mukill**
gang gang gang

### CODE REVIEW

File: src/screens/ConnectScreen.js [hunk 1]
```diff
@@ -192,8 +195,19 @@ export default class ConnectScreen extends React.Component {
     };
   }
 
+
+  onRegionChangeComplete(region) {
+    const vets = this.state.veterans.filter(vet => {
+      return parseFloat(vet.lat) <= (region.latitude + region.latitudeDelta/2) && 
+      parseFloat(vet.lat) >= (region.latitude - region.latitudeDelta/2) &&
```
Comment (jasonzbao) [inline src/screens/ConnectScreen.js]: can you indent this to the line above and do a similar thing for pos

File: src/screens/ConnectScreen.js [hunk 2]
```diff
@@ -192,8 +195,19 @@ export default class ConnectScreen extends React.Component {
...
 
+
+  onRegionChangeComplete(region) {
+    const vets = this.state.veterans.filter(vet => {
+      return parseFloat(vet.lat) <= (region.latitude + region.latitudeDelta/2) && 
+      parseFloat(vet.lat) >= (region.latitude - region.latitudeDelta/2) && 
+      parseFloat(vet.lng) <= (region.longitude + region.longitudeDelta/2) &&
+      parseFloat(vet.lng) >= (region.longitude - region.longitudeDelta/2)});
+    const pos = this.state.parterOrgs.filter(vet => {return parseFloat(vet.lat) <= region.latitude + region.latitudeDelta/2 && parseFloat(vet.lat) >= region.latitude - region.latitudeDelta/2 && parseFloat(vet.lng) <= region.longitude + region.longitudeDelta/2 && parseFloat(vet.lng) >= region.longitude - region.longitudeDelta/2});
+    this.setState({ region: region, currentPos: pos, currentVets:vets });
```
Comment (jasonzbao) [inline src/screens/ConnectScreen.js]: `currentVets: vets`

File: src/screens/ConnectScreen.js [hunk 3]
```diff
@@ -321,11 +335,20 @@ export default class ConnectScreen extends React.Component {
    * rocket function to bind the onPress method
    */
   renderVeteranMarkers() {
-    return this.state.veterans.map((veteran) => {
+    const locations = new Set();
+    let currentVets = this.state.currentVets;
+    if (this.state.currentVets.length + this.state.currentPos.length > 10) {
+      currentVets = this.state.currentVets.slice(0, 10 - this.state.currentPos.length)
```
Comment (jasonzbao) [inline src/screens/ConnectScreen.js:344]: This is not the best

File: src/screens/ConnectScreen.js [hunk 4]
```diff
@@ -321,11 +335,20 @@ export default class ConnectScreen extends React.Component {
...
+    }
+    return currentVets.map((veteran) => {
       const coordinate = {
         latitude: parseFloat(veteran.lat),
         longitude: parseFloat(veteran.lng),
       };
+      if (locations.has(coordinate)) {
```
Comment (jasonzbao) [inline src/screens/ConnectScreen.js]: Add a comment for why this is here please!

File: src/screens/ConnectScreen.js [hunk 5]
```diff
@@ -338,17 +361,47 @@ export default class ConnectScreen extends React.Component {
     });
   }
 
+
+  comparePO(a,b) {
```
Comment (jasonzbao) [inline src/screens/ConnectScreen.js]: `comparePO(a, b)`

File: src/screens/ConnectScreen.js [hunk 6]
```diff
@@ -338,17 +361,47 @@ export default class ConnectScreen extends React.Component {
...
    * rocket function to bind the onPress method
    */
   renderParterOrgMarkers() {
-    return this.state.parterOrgs.map((org) => {
+    const locations = new Set();
+    let currentPos = this.state.currentPos;
+    if (this.state.currentPos.length > 10) {
+      currentPos.sort(comparePO);
+      currentPos = currentPos.slice(0, 10);
+
```
Comment (jasonzbao) [inline src/screens/ConnectScreen.js]: remove space

File: src/screens/Find Results [hunk 7]
```diff
@@ -0,0 +1,274 @@
+Searching 36063 files for "subscribers"
```
Comment (jasonzbao) [inline src/screens/Find Results]: this gave me suicide

**Review by jasonzbao — APPROVED** [review]

File: src/screens/ConnectScreen.js [hunk 8]
```diff
@@ -192,8 +195,22 @@ export default class ConnectScreen extends React.Component {
...
+      return parseFloat(vet.lat) <= region.latitude + region.latitudeDelta/2 && 
+             parseFloat(vet.lat) >= region.latitude - region.latitudeDelta/2 &&
+             parseFloat(vet.lng) <= region.longitude + region.longitudeDelta/2 &&
+             parseFloat(vet.lng) >= region.longitude - region.longitudeDelta/2});
+    this.setState({ region: region, currentPos: pos, currentVets: vets });
+  }
+
   onRegionChange(region) {
-    this.setState({ region: region });
+    //this.setState({ region: region});
```
Comment (lbkchen) [inline src/screens/ConnectScreen.js:213]: if you comment this out then the region state is not going to be updated upon scrolling the screen so idk if this is exactly what you want

File: src/screens/ConnectScreen.js [hunk 9]
```diff
@@ -192,8 +195,22 @@ export default class ConnectScreen extends React.Component {
...
+      return parseFloat(vet.lat) <= (region.latitude + region.latitudeDelta/2) && 
+             parseFloat(vet.lat) >= (region.latitude - region.latitudeDelta/2) && 
+             parseFloat(vet.lng) <= (region.longitude + region.longitudeDelta/2) &&
+             parseFloat(vet.lng) >= (region.longitude - region.longitudeDelta/2)});
+    const pos = this.state.parterOrgs.filter(vet => {
+      return parseFloat(vet.lat) <= region.latitude + region.latitudeDelta/2 && 
+             parseFloat(vet.lat) >= region.latitude - region.latitudeDelta/2 &&
+             parseFloat(vet.lng) <= region.longitude + region.longitudeDelta/2 &&
+             parseFloat(vet.lng) >= region.longitude - region.longitudeDelta/2});
+    this.setState({ region: region, currentPos: pos, currentVets: vets });
```
Comment (lbkchen) [inline src/screens/ConnectScreen.js:209]: can you do a `parseFloat` once here in each of these methods?

File: src/screens/ConnectScreen.js (hunk 3 above)
Comment (lbkchen) [reply src/screens/ConnectScreen.js:344]: we should probably have a global constant variable so you don't have 10s everywhere in the code

File: src/screens/ProfileScreen.js [hunk 10]
```diff
@@ -215,9 +216,9 @@ export default class ProfileScreen extends React.Component {
     } else if (params.sent_friend_request || params.is_subscribed_to || this.state.sentConnectRequest) {
       return (
         <Button
-          style={margins.marginTop.md}
+          style={[margins.marginTop.md, {backgroundColor: colors.gray}]}
           onPress={connectMethod}
-          text="CONNECT"
+          text={params.is_subscribed_to ? "FOLLOWING" : "REQUESTED"}
```
Comment (lbkchen) [inline src/screens/ProfileScreen.js:221]: what if the user has not sent a request to this person yet?

**Review by lbkchen — COMMENTED** [review]
🎉

File: src/screens/ProfileScreen.js (hunk 10 above)
Comment (jasonzbao) [reply src/screens/ProfileScreen.js:221]: I think if you view the whole file, its under these lines

**Pull request merged**